GPU parity container build broken against ztensor v1.16.0 (undefined CUDA kernel symbols under -tags cuda + CGO_ENABLED=0)

## Problem

`tests/parity/Containerfile` no longer builds against the pinned `github.com/zerfoo/ztensor v1.16.0`. The build stage runs:

```
CGO_ENABLED=0 GOOS=linux GOARCH=arm64 go test -c -tags cuda -trimpath -o /out/parity-test ./tests/parity/
```

and fails with a wall of undefined kernel symbols, e.g.:

```
ztensor@v1.16.0/internal/gpuapi/cuda_kernels.go:109:17: undefined: kernels.GemmQ4F32
...:116:18: undefined: kernels.GemvQ4KSm121F32
internal/gpuapi/cuda_kernels.go:109:17: undefined: kernels.GemmQ4F32
internal/gpuapi/cuda_kernels.go:157:17: undefined: kernels.Transpose2D
internal/gpuapi/cuda_kernels.go:181:17: undefined: kernels.FusedRoPEF32
... too many errors
```

The symbols exist only under the `cutlass` tag (and/or CGO). `scripts/dgx-spark-parity.sh`, which runs the same tests on the DGX host, uses `-tags cuda,cutlass` with the host CUDA toolchain — so the host path compiles while the **container** path (`-tags cuda` + `CGO_ENABLED=0`, relying on purego/dlopen) does not. It appears ztensor's purego kernel bindings gained new Q4/Q5/Q6 GEMV + transpose/gather/rmsnorm/rope symbols that the `cuda`-only (no-cutlass, no-cgo) build does not satisfy.

## Impact

The `ghcr.io/zerfoo/zerfoo-parity` image (consumed by `docs/bench/manifests/gpu-parity.yaml`) cannot be rebuilt from current `main`, so the Spark GPU-parity path on GB10 is blocked. GPU parity must currently be run via `dgx-spark-parity.sh` on the host instead.

## Repro

```
podman build --platform linux/arm64 -f tests/parity/Containerfile -t zerfoo-parity:test .
```

## Suggested fix

Reconcile the Containerfile build tags with what ztensor v1.16.0 requires for a CGO-free build — either add the `cutlass` tag and the cutlass headers to the build stage, or restore the purego symbol set so `-tags cuda` + `CGO_ENABLED=0` links again. Whichever matches the intended "CUDA loaded at runtime via dlopen" design of the parity image.

## Context

Surfaced while building the parity image from branch `fix/t118-ondevice-gradnorm` (PR #901, T11.8) to run `TestGPUParity_GradNormReduceAll` on GB10. The fix itself is CPU-validated and CI-green; only the GPU-parity *execution* is blocked by this container-build issue.

--- a/tests/parity/Containerfile
+++ b/tests/parity/Containerfile
@@ -1,14 +1,22 @@
 # GPU parity test container.
 #
-# Multi-stage build: compiles a static test binary in the builder stage,
+# Multi-stage build: compiles the parity test binary in the builder stage,
 # then copies it into a minimal Ubuntu 24.04 runtime image. CUDA libs are
 # mounted from the host at runtime (see gpu-parity.yaml manifest).
 #
 # Build:
 #   docker build --platform linux/arm64 -f tests/parity/Containerfile -t ghcr.io/zerfoo/zerfoo-parity:latest .
 #
-# The test binary is built with CGO_ENABLED=0 — GPU acceleration is loaded
-# at runtime via purego/dlopen from host-mounted /usr/local/cuda.
+# CGO is ENABLED but the build needs NO CUDA toolkit or cutlass headers
+# (issue #902). The binary is built WITHOUT the `cuda` build tag, so:
+#   - internal/cuda selects the cgo dlopen path (purego_linux_arm64_cgo.go),
+#     which resolves dlopen/dlsym at link time (the CGO-free !cgo path's
+#     //go:linkname runtime.dlopen is unresolvable in a static binary --
+#     the original `-tags cuda` + CGO_ENABLED=0 combination did not link).
+#   - ztensor selects its `!cuda` purego kernel bindings, which dlopen the
+#     real CUDA kernels at runtime from the host-mounted /usr/local/cuda and
+#     /opt/zerfoo/lib (see gpu-parity.yaml). No CUDA/cutlass at build time.
+# A plain gcc (already in golang:1.26-bookworm) satisfies CGO.
 
 # ── Build stage ──────────────────────────────────────────────────────────────
 FROM golang:1.26-bookworm AS build
@@ -21,9 +29,10 @@ RUN go mod download
 
 COPY . .
 
-# Compile the parity test binary for linux/arm64.
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=arm64 \
-    go test -c -tags cuda -trimpath \
+# Compile the parity test binary for linux/arm64. CGO on, no `cuda` tag
+# (issue #902): cgo dlopen path links; CUDA kernels load at runtime via dlopen.
+RUN CGO_ENABLED=1 GOOS=linux GOARCH=arm64 \
+    go test -c -trimpath \
     -o /out/parity-test ./tests/parity/
 
 # ── Runtime stage ────────────────────────────────────────────────────────────

